edgexfoundry/device-sdk-c
# feat!: load common configuration from config provider

fix: #481 







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

Non-secure mode testing
1. From the latest compose builder run `make run no-secty`.
2. Run device-random example (/src/c/examples/random) with `-cp=consul.http://localhost:8500 --registry` arguments.
3. Verify that the device-random configuration has been written to Consul K/V Store.
4. Update Writable configuration for device-random and core-common-config-bootstrapper on Consul K/V Store, and then verify if it takes effect.

Secure mode testing
1. From the latest compose builder, modify add-security.yml to add secret store token and registry ACL role for `device-random`, and then run `make run`.
2. Set environment variable `EDGEX_SECURITY_SECRET_STORE` to true and run device-random example (/src/c/examples/random) with `-cp=consul.http://localhost:8500 --registry` arguments.
3. Verify that the device-random configuration has been written to Consul K/V Store.
4. Update Writable configuration for device-random and core-common-config-bootstrapper on Consul K/V Store, and then verify if it takes effect.

## New Dependency Instructions (If applicable)
